packages/trace-viewer/src/snapshotRenderer.ts

```diff
@@ -25,6 +25,34 @@ function isSubtreeReferenceSnapshot(n: NodeSnapshot): n is SubtreeReferenceSnaps
   return Array.isArray(n) && Array.isArray(n[0]);
 }
 
+let cacheSize = 0;
+const cache = new Map<SnapshotRenderer, string>();
+const CACHE_SIZE = 300000000; // 300mb
+
+function cacheAndReturn(key: SnapshotRenderer, compute: () => string): string {
+  if (cache.has(key)) {
+    const value = cache.get(key)!;
+    // reinserting makes this the least recently used entry
+    cache.delete(key);
+    cache.set(key, value);
+    return value;
+  }
+
+
+  const result = compute();
+
+  while (cacheSize + result.length > CACHE_SIZE) {
```

Comment:
To avoid npe when `result.length > CACHE_SIZE`.

```suggestion
  while (cache.size && cacheSize + result.length > CACHE_SIZE) {
```